test/functional/address_types.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test that the wallet can send and receive using all combinations of address types."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class AddressTypeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 4
+        self.setup_clean_chain = True
+        self.extra_args = [["-addresstype=legacy"], ["-addresstype=p2sh"], ["-addresstype=bech32"], ["-addresstype=p2sh", "-changetype=bech32"]]
+
+    def run_test(self):
+        self.nodes[0].generate(120)
+        self.sync_all()
+
+        for c in range(4):
+            for n in range(4):
+                old_balances = [self.nodes[i].getbalance() for i in range(4)]
+                to_send = (old_balances[n] / 11).quantize(Decimal("0.00000001"))
+                sends = {}
+                for k in range(4):
+                    address = self.nodes[(n + k) % 4].getnewaddress()
+                    # In each iteration, one node sends 1/11th of its balance to itself, 2/11ths to the next peer,
+                    # 3/11ths to the one after that, and 4/11ths to the remaining one.
+                    sends[address] = to_send * (1 + k)
+                self.nodes[n].sendmany("", sends)
+                self.sync_all()
+                unconf_balances = [self.nodes[i].getunconfirmedbalance() for i in range(4)]
```

Comment:
Should `getunconfirmedbalance` be the same as what QT displays as balance under "Pending"? If so, what could explain that the UI shows different numbers despite these tests passing?